### PR TITLE
[CDF-985] Fixed the issue when not escaped html value for text …

### PR DESCRIPTION
**Pull request by annapopova1**
…input component breaks it.

For example, the value is '<>' and as a result we had html like <div id="46205-98607"><input type="text" id="46205-98607-input" name="46205-98607-input" value="" <="">''&gt;</div>

@afrjorge @pentaho-arua @eliofreitas @nunodelisboa please review

### CODE REVIEW

File: core-js/src/main/javascript/cdf/components/TextInputComponent.js [hunk 1]
```diff
@@ -20,30 +20,18 @@ define([
...
-      if(myself.maxLength) {
-        Logger.warn("Attribute 'maxLength' is deprecated");
-      }
-
-      myself.placeholder().html(selectHTML);
-
-      var el = $("#" + name);
+      myself._addHtmlToPlaceholder();
 
+      var el = $("#" + this.name);
```
Comment (nunodelisboa) [inline core-js/src/main/javascript/cdf/components/TextInputComponent.js]: Isn't this `this` supposed to be `myself`?

File: core-js/src/main/javascript/cdf/components/TextInputComponent.js [hunk 2]
```diff
@@ -20,30 +20,18 @@ define([
...
+      el.val(this.dashboard.getParameterValue(this.parameter));
```
Comment (nunodelisboa) [inline core-js/src/main/javascript/cdf/components/TextInputComponent.js]: Arent't these `this` themselves supposed to be `myself` ?

File: core-js/src/main/javascript/cdf/components/TextInputComponent.js (hunk 1 above)
Comment (annapopova1) [inline core-js/src/main/javascript/cdf/components/TextInputComponent.js]: in this line it's the same. myself makes sense for callbacks

File: core-js/src/main/javascript/cdf/components/TextInputComponent.js (hunk 2 above)
Comment (annapopova1) [inline core-js/src/main/javascript/cdf/components/TextInputComponent.js]: the same as above

File: core-js/src/main/javascript/cdf/components/TextInputComponent.js [hunk 3]
```diff
@@ -52,6 +40,19 @@ define([
 
     getValue: function() {
       return $("#" + this.name).val();
+    },
+
+    _addHtmlToPlaceholder: function() {
```
Comment (nunodelisboa) [inline core-js/src/main/javascript/cdf/components/TextInputComponent.js]: Isn't it easier to read if it is something like
```
function() {
  var attrs = { 
    type: "text",
    id: this.name,
    name: this.name,
    size: this.size || this.charWidth || undefined,
    maxlength: this.maxLength || this.maxCharts || undefined
};

var componentHTML = '<input ' + processAttrs(attrs) + '>';

if (this.size) Logger.warn("Attribute 'size' is deprecated");
if (this.maxLength) Logger.warn("Attribute 'maxLength' is deprecated");

this.placeholder().html(componentHTML);

function processAttrs(conf) {
  var list = [];
  for (prop in conf){
    if (conf.hasOwnProperty(prop). && conf[prop] != null){
      list.push(prop + '="' + conf[prop] + '"');
    }
  }
   return list.join(" ");
 }
}
```

Comment (nunodelisboa) [inline core-js/src/main/javascript/cdf/components/TextInputComponent.js]: Also, maybe use the `sanitiser` on `conf[prop]` just to be safe

**Pull request merged**